src/services/storage.js

```diff
@@ -1,9 +1,33 @@
 import {createState} from '@hookstate/core'
 
-const starCount = createState(5);
+const starCount = createState(0);
+const currentIndex = createState(null);
+const favouriteMovies = createState([]);
 
-const increaseStarCount = () => {
-    starCount.set(p => p + 1);
+const addFavourite = (movie) => {
+    favouriteMovies.set([...favouriteMovies, movie]);
+    console.log(favouriteMovies.get());
+};
+
+const removeFavourite = (movie) => {
+    const filteredList = favouriteMovies.filter(
+        item => item !== movie
+      );
+      favouriteMovies.set(filteredList);
+}
+
+
+const increaseStarCount = (id) => {
+    
+    if(id == currentIndex.get()){
+        starCount.set(p => p - 1);
+        currentIndex.set(null);
+        removeFavourite(id);
+    } else {
+        starCount.set(p => p + 1);
+        currentIndex.set(id);
+        addFavourite(id);
+    }
 }
```

Comment:
The function that increasing StarCount on click. It takes the id of the movie as a parameter and controls which movie was recently clicked.